File: flutter_basics/task_manager_backend_frontend/task_manager_api/models/Task.php
<?php

namespace app\models;

use Yii;
use yii\behaviors\TimestampBehavior;
use yii\db\ActiveRecord;

/**
 * This is the model class for table "task".
 *
 * @property int $id
 * @property string $title
 * @property string|null $description
 * @property string|null $status
 * @property int $created_at
 * @property int $updated_at
 */
class Task extends ActiveRecord
{
    /**
     * {@inheritdoc}
     */
    public static function tableName()
    {
        return '{{%task}}';
    }

    /**
     * {@inheritdoc}
     */
    public function behaviors()
    {
        return [
            TimestampBehavior::class,
        ];
    }

    /**
     * {@inheritdoc}
     */
    public function rules()
    {
        return [
            [['title'], 'required'],
            [['description'], 'string'],
            [['created_at', 'updated_at', 'priority'], 'integer'],
            [['title', 'status', 'assigned_to'], 'string', 'max' => 255],
            ['status', 'default', 'value' => 'pending'],
            ['status', 'in', 'range' => ['pending', 'in_progress', 'suspended', 'to_release', 'completed']],
            ['priority', 'default', 'value' => 1],
            ['priority', 'integer', 'min' => 1, 'max' => 5],
        ];
    }

    /**
     * {@inheritdoc}
     */
    public function attributeLabels()
    {
        return [
            'id' => 'ID',
            'title' => 'Title',
            'description' => 'Description',
            'status' => 'Status',
            'assigned_to' => 'Assigned To',
            'priority' => 'Priority',
            'created_at' => 'Created At',
            'updated_at' => 'Updated At',
        ];
    }
}

File: flutter_basics/task_manager_backend_frontend/task_manager_frontend/lib/models/task.dart
class Task {
  final int? id;
  final String title;
  final String? description;
  final String status;
  final String? assignedTo;
  final int priority;
  final int? createdAt;
  final int? updatedAt;

  Task({
    this.id,
    required this.title,
    this.description,
    this.status = 'pending',
    this.assignedTo,
    this.priority = 1,
    this.createdAt,
    this.updatedAt,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      status: json['status'],
      assignedTo: json['assigned_to'],
      priority: json['priority'] ?? 1,
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'title': title,
      if (description != null) 'description': description,
      'status': status,
      if (assignedTo != null) 'assigned_to': assignedTo,
      'priority': priority,
    };
  }

  Task copyWith({
    int? id,
    String? title,
    String? description,
    String? status,
    String? assignedTo,
    int? priority,
    int? createdAt,
    int? updatedAt,
  }) {
    return Task(
      id: id ?? this.id,
      title: title ?? this.title,
      description: description ?? this.description,
      status: status ?? this.status,
      assignedTo: assignedTo ?? this.assignedTo,
      priority: priority ?? this.priority,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}
